### PR TITLE
Fix error propagation for checkTx errors

### LINKED ISSUE #800: bug: New broadcast logic hides errors
The latest implementation of broadcastTx that polls the chain for the tx result hides all pre-commit errors.

https://github.com/cosmos/cosmjs/blob/main/packages/stargate/src/stargateclient.ts#L329

This will return a timeout error for any kind of invalid transaction including insufficient fees. etc.

### PR DESCRIPTION
**Pull request by webmaster128**
Closes #800

### CODE REVIEW

File: packages/cosmwasm-stargate/src/cosmwasmclient.ts [hunk 1]
```diff
@@ -240,10 +251,15 @@ export class CosmWasmClient {
...
+      throw new Error(
+        `Broadcasting transaction failed with code ${broadcasted.code} (codespace: ${broadcasted.codeSpace}). Log: ${broadcasted.log}`,
+      );
+    }
+    const transactionId = toHex(broadcasted.hash).toUpperCase();
     return new Promise((resolve, reject) =>
-      this.forceGetTmClient()
-        .broadcastTxSync({ tx })
-        .then(({ hash }) => pollForTx(toHex(hash).toUpperCase()))
+      pollForTx(transactionId)
```
Comment (webmaster128) [inline packages/cosmwasm-stargate/src/cosmwasmclient.ts:262]: I'm still unsure if the use of

```
        .then(resolve, reject)
        .finally(() => clearTimeout(txPollTimeout)),
```

is correct. Shoudn't the call of `resolve` or `resolve` be the last thing that happens here?

Comment (willclarktech) [reply packages/cosmwasm-stargate/src/cosmwasmclient.ts:262]: Happy to move the `clearTimeout` into the `then` handlers to make it clearer.

File: packages/stargate/src/stargateclient.ts [hunk 2]
```diff
@@ -99,6 +99,15 @@ export interface BroadcastTxSuccess {
   readonly gasWanted: number;
 }
 
+/**
+ * The response after sucessfully broadcasting a transaction.
```
Comment (willclarktech) [inline packages/stargate/src/stargateclient.ts:103]:
```suggestion
 * The response after successfully broadcasting a transaction.
```

File: packages/stargate/src/stargateclient.ts [hunk 3]
```diff
@@ -292,6 +301,17 @@ export class StargateClient {
...
+  /**
+   * Broadcasts a signed transaction to the network and monitors its inclusion in a block.
+   *
+   * If broadcasting is rejected by the node for some reason (e.g. because of a CheckTx failure),
+   * an error is thrown.
+   *
+   * If the transaction is not included in a block before the provided timeout, this errors with a `TimeoutError`.
+   *
+   * If the transaction is included in a block, a `BroadcastTxResponse` is returned. The caller then
+   * usually needs check for execution success or failure.
```
Comment (willclarktech) [inline packages/stargate/src/stargateclient.ts:313]:
```suggestion
   * usually needs to check for execution success or failure.
```

**Review by willclarktech — APPROVED** [review]

File: packages/cosmwasm-stargate/src/cosmwasmclient.ts (hunk 1 above)
Comment (webmaster128) [reply packages/cosmwasm-stargate/src/cosmwasmclient.ts]: Done in 33ff475. This changes the order of execution such that nothing happens after the resolve/reject call.

**Pull request merged**